### ISSUE TITLE
S40 firmware 4.27.0054 - Exception in _processEquipments

### ISSUE THREAD

**Issue by gdoerr** [PeteRager/lennoxs30#321]
**Describe the bug**
With the newest version of the S40 software, it appears that the format of some of the JSON has changed from a single "value" to an array of values (example of the new JSON below). This causes an exception.

This causes the enumeration of equipment to fail and many entities to be Unavailable.

**To Reproduce**
Upgrade thermostat to the latest Lennox version (4.27.0054)

**Expected behavior**
The code should accept both legacy and new "values" JSON data formats.

**Screenshots/Logs**
```
File "/usr/local/lib/python3.12/site-packages/lennoxs30api/s30api_async.py", line 1654, in _processEquipments
eq.equipment_name = parameter["parameter"]["value"]
~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'value'

```

JSON snippet:
```
            {
              "id": 1,
              "feature": {
                "name": "Equipment Type Name",
                "format": "nts",
                "szValues": 1,
                "values": [
                  {
                    "id": 0,
                    "value": "Air Conditioner"
                  }
                ],
                "fid": 15,
                "unit": ""
              }
            },
            {
              "id": 2,
              "feature": {
                "name": "Unit Model Number",
                "format": "nts",
                "szValues": 1,
                "values": [
                  {
                    "id": 0,
                    "value": "SL28XCV048-230A01"
                  }
                ],
                "fid": 6,
                "unit": ""
              }
            },
```

**Version**
 - Home Assistant: 2024.4.3
 - This Integration: 2024.3.0
 

**Integration Configuration **
- Cloud or Local Connection: Local
- Inverter or Sensor Diagnostics Enabled: Both are enabled

**Additional context**
None

**Comment by PeteRager** [comment]
Please post the message log file.  Maybe delete the current one and do an HA restart, let it run for 5.

**Comment by gdoerr** [comment]
As requested
[S30_msg.log](https://github.com/PeteRager/lennoxs30/files/14986310/S30_msg.log)

**Comment by PeteRager** [comment]
None of the equipment parameters for eqtype=18 (air conditioner) have any values and have a descriptor of "error".  The other equipment seems fine, so it's just that one unit.  

While I can and will update the software to not crash when this occurs.  The equipment name and the parameters for the air conditioner are not available.  The diagnostics look ok.

I would try power cycling everything.   The S40 panels, indoor units, outdoor unit.  To see if that corrects the problem.  


Here is the excerpt from your message.

```
                            "parameters": [
                                {
                                    "parameter": {
                                        "name": "Equipment Name",
                                        "format": "nts",
                                        "pid": 18,
                                        "enabled": true,
                                        "descriptor": "error",
                                        "unit": ""
                                    },
                                    "id": 0
                                },
                                {
                                    "parameter": {
                                        "name": "Compressor Short Cycle Delay",
                                        "format": "uint16",
                                        "pid": 40,
                                        "enabled": true,
                                        "descriptor": "error",
                                        "unit": "sec"
                                    },
                                    "id": 1
                                },
                                {
                                    "parameter": {
                                        "name": "High Normal Cooling Airflow",
                                        "format": "uint16",
                                        "pid": 44,
                                        "enabled": true,
                                        "descriptor": "error",
                                        "unit": "CFM"
                                    },
                                    "id": 2
                                },
                                {
                                    "parameter": {
                                        "name": "Low Normal Cooling Airflow",
                                        "format": "uint16",
                                        "pid": 45,
                                        "enabled": true,
                                        "descriptor": "error",
                                        "unit": "CFM"
                                    },
                                    "id": 3
                                },
                                {
                                    "parameter": {
                                        "name": "Dehum Airflow Adjustment Adder",
                                        "format": "0p5",
                                        "pid": 24,
                                        "enabled": true,
                                        "descriptor": "error",
                                        "unit": "%"
                                    },
                                    "id": 4
                                }
                            ],
```

**Comment by PeteRager** [comment]
https://github.com/PeteRager/lennoxs30api/pull/80

**Comment by PeteRager** [comment]
Updates on this item?

**Comment by gdoerr** [comment]
The exception is resolved with the API change, thank you. I have rebooted the indoor, outdoor and thermostat units but I'm still not getting good data. 

I have time this weekend and will try a full power down again (have to climb into the attic so it's a bit of a pain)

**Comment by PeteRager** [comment]
Thanks for the update.  You may be able to toss the breakers.  Mine has a 30amp for outdoor unit and a 20 for the air handler.

**Comment by PeteRager** [comment]
Released in 2024.6.0

**Comment by PeteRager** [comment]
@gdoerr have a user with a similar problem of not seeing the outdoor unit, did this problem resolve for you? What was the fix?

**Comment by gdoerr** [comment]
Hi Pete,

The root of my issue ended up being that I have a phantom outdoor unit (something weird the installer did). Before the fix you made, the integration would crash since there was limited data coming in from the unit. That wasn't obvious until your analysis caused me to take a deeper look at the data coming back from Lennox.

After a lengthy power cycle I did start getting data back. I did have to turn power off to both the AC and Furnace units and leave the power off for a good 15 minutes before restoring power (AC first, then Furnance). Not sure if the order matters but this is what worked for me.
